crates/sui-core/src/consensus_validator.rs

```diff
@@ -68,9 +68,26 @@ impl TransactionValidator for SuiTxValidator {
 
         // let mut owned_tx_certs = Vec::new();
         let mut obligation = VerificationObligation::default();
+        let mut cached_cert_digests = Vec::new();
+        let mut verified_cert_digests = Vec::new();
         for tx in txs.into_iter() {
             match tx.kind {
                 ConsensusTransactionKind::UserTransaction(certificate) => {
+                    let cert_digest = certificate.certificate_digest();
+                    if self
+                        .epoch_store
+                        .verified_cert_cache()
+                        .is_cert_verified(&cert_digest)
+                    {
+                        cached_cert_digests.push(cert_digest);
+                        self.metrics.certificate_signatures_cache_hits.inc();
+                        continue;
```

Comment:
i think so - i will write some tests to confirm

Reply:
assuming our validator should not accept invalid signed transaction even if >f validators are malicious (as full nodes should), we need to verify those signatures even if they are part of the certificate

Reply:
General rule of thumb, similarly to Ethereum and other BFT threat models, typically each honest validator does not blindly trust what they see (even if signed by a quorum) and they need to also execute everything on their side as well, so yes user sig is also part of this work (otherwise you are technically a light node).

Reply:
I have written a test for this, the digest definitely includes user sigs, so presence in the cache proves that we verified the user sigs previously. Here we will verify the sigs on any un-cached certs on line 92.

Reply:
This is true now because we insert to the cache only after we verified the tx signatures. If it was the opposite direction for some reason (not sure why), and the tx sig verification had failed, the cache would still include the cert. Perhaps let's mention this "assumption" (that tx sigs are verified before) in a comment in authority server?

Reply:
> This is true now because we insert to the cache only after we verified the tx signatures.

yes but anything other than that would be plainly incorrect? I will add a comment to clarify though.